### PR TITLE
added a case in trie remove

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/trie.cpp [hunk 1]
```diff
@@ -158,7 +158,7 @@ bool Trie::_remove(TrieNode *current, string str, int index)
     bool shouldDeleteCurrentNode = _remove(current->children[ch], str, index + 1);
 
     // after recursion
-    if (shouldDeleteCurrentNode)
+    if (shouldDeleteCurrentNode&&!current->childeren[ch]->isEndOfWord)
```
Comment (attnishit) [inline src/trie.cpp:161]: This is a redundant condition and need not to be checked.

Comment (RohitKaushal7) [reply src/trie.cpp:161]: see line 141 - if `shouldDeleteCurrentNode`  is true then  `children[ch]->isEndOfWord`  have to be false.

Comment (RohitKaushal7) [reply src/trie.cpp:161]: see line 141 - if `shouldDeleteCurrentNode` is true then `children[ch]->isEndOfWord` have to be false.

Comment (ghost) [reply src/trie.cpp:161]: > see line 141 - if `shouldDeleteCurrentNode` is true then `children[ch]->isEndOfWord` have to be false.

but that only checks when `index ==str.length()`
for eg in picture you shared if we delete or remove _themes_; _theme_ is also deleted which should not happen. only _thus_ is available 
[pic](https://camo.githubusercontent.com/33bf04613b8b022f24875e4a27c73f254ec43a4c63a6397218826e5b86c40528/68747470733a2f2f342e62702e626c6f6773706f742e636f6d2f2d386b347768375548676b302f57586d4c486c64445a79492f414141414141414141436f2f4b6749324c6b4a4b6169517a4d436563783931644f4672534b6343645744786867434c63424741732f733634302f747269652e706e67)

**Review by RohitKaushal7 — CHANGES_REQUESTED** [review]
Yes your test case is right. thanks.
just a minor change, instead of this
```cpp
if (shouldDeleteCurrentNode&&!current->childeren[ch]->isEndOfWord)
```
do this in line 164
```cpp
return !current->isEndOfWord && current->children.size() == 0; // true ?
```
so it becomes 
```cpp
// after recursion
if (shouldDeleteCurrentNode)
{
    current->children.erase(ch);
    return !current->isEndOfWord && current->children.size() == 0; // true ?
}

**Comment by ghost** [comment]
thanks for responding so fast. i thought u will never respond :)

**Pull request closed without merging**